src/Xamarin.Android.Build.Tasks/Utilities/Aapt2Daemon.cs

```diff
@@ -162,21 +165,34 @@ private void Aapt2DaemonStart ()
 				// and we are using netstandard 2.0
 				//StandardInputEncoding = Encoding.UTF8,
 			};
-			// We need to FORCE the StandardInput to be UTF8 so we can use 
+			// We need to FORCE the StandardInput to be UTF8 so we can use
 			// accented characters. Also DONT INCLUDE A BOM!!
 			// otherwise aapt2 will try to interpret the BOM as an argument.
-			Process aapt2;
+			Process aapt2 = null;
 			lock (lockObject) {
-				Encoding current = Console.InputEncoding;
 				try {
-					Console.InputEncoding = new UTF8Encoding (false);
-					aapt2 = new Process ();
-					aapt2.StartInfo = info;
-					aapt2.Start ();
-				} finally {
-					Console.InputEncoding = current;
+					Encoding current = Console.InputEncoding;
+					try {
+						try {
+							Console.InputEncoding = new UTF8Encoding (false);
+						} catch (IOException ioEx) {
+							// For some reason we can not set the InputEncoding. If this happens we don't
+							// want to crash Visual Studio with an Unhandled Exception.
+							// The downside of ignoring this is paths with accented characters will cause problems.
+							daemonStartupWarnings.Enqueue (ioEx.ToString ());
+						}
+						aapt2 = new Process ();
+						aapt2.StartInfo = info;
+						aapt2.Start ();
+					} finally {
+						Console.InputEncoding = current;
+					}
+				} catch (Exception ex) {
+					daemonStartupWarnings.Enqueue (ex.ToString ());
 				}
 			}
+			if (aapt2 == null)
```

Comment:
How can this happen?  Line 184 will re-assign `aapt2` to a non-null value, unless the `Process` constructor throws.

If `aapt2` *could* be null, that sounds like a possible "semantic" violation, as other parts of the code will assume/require that `aapt2` be running, but if it *isn't* running, what'll happen?